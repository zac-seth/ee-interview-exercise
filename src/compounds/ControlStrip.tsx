import { FC } from 'react';

import {
  CalculateButton,
  HandleCalculateSelected,
  HandleResetSelected,
  HorizontalStrip,
  PlaceholderButton,
  ResetButton,
} from 'elements';

export type ControlStripProps = {
  onCalculateSelected: HandleCalculateSelected;
  onResetSelected: HandleResetSelected;
};

export const ControlStrip: FC<ControlStripProps> = ({
  onCalculateSelected,
  onResetSelected
}) => (
  <HorizontalStrip length={4}>
    <PlaceholderButton label='?' />
    <ResetButton
      onClick={() => onResetSelected({ isFullReset: true })}
      label='CE'
    />
    <ResetButton onClick={onResetSelected} label='C' />
    <CalculateButton onClick={onCalculateSelected} label='=' />
  </HorizontalStrip>
);
