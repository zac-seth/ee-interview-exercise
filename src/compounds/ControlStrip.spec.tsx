import { fireEvent, render, screen } from "@testing-library/react";

import { ControlStrip } from "./ControlStrip";

const handleCalculateSelected = jest.fn();
const handleResetSelected = jest.fn();

describe(`ControlStrip`, () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it(`Should render the control button strip`, () => {
    render(
      <ControlStrip onCalculateSelected={() => {}} onResetSelected={() => {}} />
    );

    expect(screen.getByTestId(`horizontal-strip-layout`)).toMatchSnapshot();
  });

  it(`Should render the correct buttons in the strip`, async () => {
    render(
      <ControlStrip onCalculateSelected={() => {}} onResetSelected={() => {}} />
    );

    expect(await screen.findByText(`?`)).toBeDefined();
    expect(await screen.findByText(`CE`)).toBeDefined();
    expect(await screen.findByText(`C`)).toBeDefined();
    expect(await screen.findByText(`=`)).toBeDefined();
  });

  it(`Should correctly pass the divide function when divide is selected`, async () => {
    render(
      <ControlStrip
        onCalculateSelected={handleCalculateSelected}
        onResetSelected={handleResetSelected}
      />
    );

    fireEvent.click(screen.getByText(`CE`));

    expect(handleResetSelected).toHaveBeenCalledTimes(1);
    expect(handleResetSelected).toHaveBeenCalledWith({ isFullReset: true });
  });

  it(`Should correctly pass the divide function when multiply is selected`, async () => {
    render(
      <ControlStrip
        onCalculateSelected={handleCalculateSelected}
        onResetSelected={handleResetSelected}
      />
    );

    fireEvent.click(screen.getByText(`C`));

    expect(handleResetSelected).toHaveBeenCalledTimes(1);
  });

  it(`Should correctly pass the divide function when subtract is selected`, async () => {
    render(
      <ControlStrip
        onCalculateSelected={handleCalculateSelected}
        onResetSelected={handleResetSelected}
      />
    );

    fireEvent.click(screen.getByText(`=`));

    expect(handleCalculateSelected).toHaveBeenCalledTimes(1);
  });
});
