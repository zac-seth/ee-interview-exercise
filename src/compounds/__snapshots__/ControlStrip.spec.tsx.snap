// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlStrip Should render the control button strip 1`] = `
.c1 {
  font-family: 'Lato',sans-serif;
  font-weight: 400;
  color: #666;
  text-shadow: 0 0 5px #1A94D2,0 0 15px #228DFF;
  font-size: 4rem;
  line-height: 4rem;
  text-align: center;
  padding: 1rem;
  margin: 0;
  width: 8rem;
  height: 8rem;
  background-color: inherit;
  color: #1a94d2;
  border: 0;
  cursor: pointer;
  color: #666;
}

.c1:hover {
  background-color: #1a1a1a;
}

.c1:hover {
  background-color: inherit;
  cursor: default;
}

.c2 {
  font-family: 'Lato',sans-serif;
  font-weight: 400;
  color: #666;
  text-shadow: 0 0 5px #1A94D2,0 0 15px #228DFF;
  font-size: 4rem;
  line-height: 4rem;
  text-align: center;
  padding: 1rem;
  margin: 0;
  width: 8rem;
  height: 8rem;
  background-color: inherit;
  color: #1a94d2;
  border: 0;
  cursor: pointer;
  background-color: #1a1a1a;
}

.c2:hover {
  background-color: #1a1a1a;
}

.c2:hover {
  background-color: #1d1d1d;
}

.c3 {
  font-family: 'Lato',sans-serif;
  font-weight: 400;
  color: #666;
  text-shadow: 0 0 5px #1A94D2,0 0 15px #228DFF;
  font-size: 4rem;
  line-height: 4rem;
  text-align: center;
  padding: 1rem;
  margin: 0;
  width: 8rem;
  height: 8rem;
  background-color: inherit;
  color: #1a94d2;
  border: 0;
  cursor: pointer;
  background-color: #1a2b1a;
}

.c3:hover {
  background-color: #1a1a1a;
}

.c3:hover {
  background-color: #1d2e1d;
}

.c0 {
  display: grid;
  grid-gap: 0.25rem;
  margin: 0;
  padding: 0;
  grid-template-columns: repeat(4,1fr);
}

<div
  class="c0"
  data-testid="horizontal-strip-layout"
>
  <button
    class="c1"
  >
    ?
  </button>
  <button
    class="c2"
  >
    CE
  </button>
  <button
    class="c2"
  >
    C
  </button>
  <button
    class="c3"
  >
    =
  </button>
</div>
`;
